Imports in .sq files containing certain SQL keywords throw SqlDelightException

Example:
```SQL
import no.test.TestEnum;

CREATE TABLE test_table (
    test_column TEXT AS TestEnum
);
```
Stacktrace:
```java
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:generateDebugSqlDelightInterface'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:68)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:81)
        at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:46)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: com.squareup.sqldelight.gradle.SqlDelightException: Example.sq line 1:7 - no viable alternative at input 'no'
        at com.squareup.sqldelight.gradle.ErrorListener.syntaxError(ErrorListener.kt:26)
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:65)
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:558)
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportNoViableAlternative(DefaultErrorStrategy.java:310)
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:147)
        at com.squareup.sqldelight.SqliteParser.java_type_name(SqliteParser.java:7048)
        at com.squareup.sqldelight.SqliteParser.import_stmt(SqliteParser.java:399)
        at com.squareup.sqldelight.SqliteParser.sql_stmt_list(SqliteParser.java:324)
        at com.squareup.sqldelight.SqliteParser.parse(SqliteParser.java:216)
        at com.squareup.sqldelight.gradle.SqlDelightTask.parseThen(SqlDelightTask.kt:112)
        at com.squareup.sqldelight.gradle.SqlDelightTask.execute(SqlDelightTask.kt:58)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.doExecute(AnnotationProcessingTaskFactory.java:245)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:221)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.execute(AnnotationProcessingTaskFactory.java:232)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:210)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
```

--- a/sqldelight-compiler/src/main/antlr/com/squareup/sqldelight/Sqlite.g4
+++ b/sqldelight-compiler/src/main/antlr/com/squareup/sqldelight/Sqlite.g4
@@ -648,7 +648,7 @@ java_type_name2
  ;
 
 java_type
- : ( IDENTIFIER '.' )* IDENTIFIER
+ : ( ( IDENTIFIER | keyword ) '.' )* ( IDENTIFIER | keyword )
  ;
 
 any_name

--- a/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/expected/com/sample/TestModel.java
+++ b/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/expected/com/sample/TestModel.java
@@ -1,0 +1,85 @@
+package com.sample;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import com.squareup.sqldelight.ColumnAdapter;
+import com.squareup.sqldelight.RowMapper;
+import java.lang.Override;
+import java.lang.String;
+import no.test.TestEnum;
+
+public interface TestModel {
+  String TABLE_NAME = "test_table";
+
+  String TEST_COLUMN = "test_column";
+
+  String CREATE_TABLE = ""
+      + "CREATE TABLE test_table (\n"
+      + "    test_column TEXT\n"
+      + ")";
+
+  @Nullable
+  TestEnum test_column();
+
+  interface Creator<T extends TestModel> {
+    T create(@Nullable TestEnum test_column);
+  }
+
+  final class Mapper<T extends TestModel> implements RowMapper<T> {
+    private final Factory<T> testModelFactory;
+
+    public Mapper(Factory<T> testModelFactory) {
+      this.testModelFactory = testModelFactory;
+    }
+
+    @Override
+    public T map(@NonNull Cursor cursor) {
+      return testModelFactory.creator.create(
+          cursor.isNull(0) ? null : testModelFactory.test_columnAdapter.map(cursor, 0)
+      );
+    }
+  }
+
+  final class Marshal {
+    protected final ContentValues contentValues = new ContentValues();
+
+    private final ColumnAdapter<TestEnum> test_columnAdapter;
+
+    Marshal(@Nullable TestModel copy, ColumnAdapter<TestEnum> test_columnAdapter) {
+      this.test_columnAdapter = test_columnAdapter;
+      if (copy != null) {
+        this.test_column(copy.test_column());
+      }
+    }
+
+    public ContentValues asContentValues() {
+      return contentValues;
+    }
+
+    public Marshal test_column(TestEnum test_column) {
+      test_columnAdapter.marshal(contentValues, TEST_COLUMN, test_column);
+      return this;
+    }
+  }
+
+  final class Factory<T extends TestModel> {
+    public final Creator<T> creator;
+
+    public final ColumnAdapter<TestEnum> test_columnAdapter;
+
+    public Factory(Creator<T> creator, ColumnAdapter<TestEnum> test_columnAdapter) {
+      this.creator = creator;
+      this.test_columnAdapter = test_columnAdapter;
+    }
+
+    public Marshal marshal() {
+      return new Marshal(null, test_columnAdapter);
+    }
+
+    public Marshal marshal(TestModel copy) {
+      return new Marshal(copy, test_columnAdapter);
+    }
+  }
+}

--- a/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/src/main/java/no/test/TestEnum.java
+++ b/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/src/main/java/no/test/TestEnum.java
@@ -1,0 +1,5 @@
+package no.test;
+
+public enum TestEnum {
+  ONE, TWO;
+}

--- a/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/src/main/sqldelight/com/sample/Test.sq
+++ b/sqldelight-gradle-plugin/src/test/fixtures/sqlite-keyword-java-package/src/main/sqldelight/com/sample/Test.sq
@@ -1,0 +1,5 @@
+import no.test.TestEnum;
+
+CREATE TABLE test_table (
+    test_column TEXT AS TestEnum
+);

